### PR TITLE
Add waiting for kyma-system namespace deletion

### PR DESCRIPTION
**Pull request by lindnerby**
**Description**
Since kyma-system namespace stays in Terminating state for some time when deleted during `undeploy`, it should wait and run until the namespace is not found anymore

Changes proposed in this pull request:

- When waiting for kyma-system namespaces to be terminated, also remove finalizers
- Times out after 4 minutes
- Add prompt for user confirmation if not `ci` and not `non-interactive`

**Related issue(s)**
* https://github.com/kyma-project/cli/issues/1006

### CODE REVIEW

File: cmd/kyma/undeploy/cmd.go [hunk 1]
```diff
@@ -297,6 +284,66 @@ func (cmd *command) deleteKymaNamespaces() error {
...
+func (cmd *command) waitForNamespaces() error {
+
+	cmd.NewStep("Waiting for namespace termination")
+
+	timeout := time.After(4 * time.Minute)
+	poll := time.Tick(5 * time.Second)
+	for {
+		select {
+		case <-timeout:
+			cmd.CurrentStep.Failuref("Timed out when waiting for deletion of kyma-system namespace")
```
Comment (NHingerl) [inline cmd/kyma/undeploy/cmd.go:296]:
```suggestion
			cmd.CurrentStep.Failuref("Timed out while waiting for deletion of kyma-system namespace")
```

File: cmd/kyma/undeploy/cmd.go [hunk 2]
```diff
@@ -297,6 +284,66 @@ func (cmd *command) deleteKymaNamespaces() error {
 	}
 }
 
+func (cmd *command) waitForNamespaces() error {
+
+	cmd.NewStep("Waiting for namespace termination")
```
Comment (NHingerl) [inline cmd/kyma/undeploy/cmd.go:289]: why is it called "termination" here? Is this something different from the "deletion" mentioned in line 272 or 296? 
If it's the same, let's say "deletion" here too.
```suggestion
	cmd.NewStep("Waiting for namespace deletion")
```

File: cmd/kyma/undeploy/cmd.go [hunk 3]
```diff
@@ -286,14 +286,14 @@ func (cmd *command) deleteKymaNamespaces() error {
...
-	cmd.NewStep("Waiting for namespace termination")
+	cmd.NewStep("Waiting for Namespace deletion")
 
 	timeout := time.After(4 * time.Minute)
 	poll := time.Tick(5 * time.Second)
 	for {
 		select {
 		case <-timeout:
-			cmd.CurrentStep.Failuref("Timed out when waiting for deletion of kyma-system namespace")
+			cmd.CurrentStep.Failuref("Timed out while waiting for deletion of Namespace")
```
Comment (NHingerl) [inline cmd/kyma/undeploy/cmd.go:296]:
```suggestion
			cmd.CurrentStep.Failuref("Timed out while waiting for Namespace deletion")
```
even shorter! :D

**Review by skhalash — APPROVED** [review]

**Pull request merged**